File: internal/engine/testdata/repos/kotlin_swift_multirepo/android/app/src/main/kotlin/de/golf/app/Analytics.kt
package de.golf.app

// Three nested types whose names are shared, by coincidence of domain vocabulary,
// with the Swift app in ../ios. The two repos share no source: each is an
// independent client of the same backend. No cross-repo edge may be drawn between
// them on the strength of these names alone.

class RegisterUseCase {
    data class ValidationError(val field: String, val reason: String)

    fun validate(email: String): ValidationError? {
        if (!email.contains("@")) {
            return ValidationError("email", "missing @")
        }
        return null
    }
}

class HandicapAnalytics {
    data class DifferentialEntry(val score: Double, val courseRating: Double)

    fun differentials(scores: List<Double>): List<DifferentialEntry> {
        return scores.map { DifferentialEntry(it, 72.0) }
    }
}

class FullAnalysisDataBuilder {
    data class TimeWindow(val startDay: Int, val endDay: Int)

    fun lastMonth(): TimeWindow = TimeWindow(0, 30)
}

File: internal/engine/testdata/repos/kotlin_swift_multirepo/ios/Sources/Core/Analytics.swift
import Foundation

// The Swift half of the parallel-app pair. The nested type names below match the
// Kotlin app in ../android because both model the same product, not because either
// repo includes the other's source.

public class RegisterUseCase {
    public struct ValidationError {
        public let field: String
        public let reason: String
    }

    public func validate(email: String) -> ValidationError? {
        if !email.contains("@") {
            return ValidationError(field: "email", reason: "missing @")
        }
        return nil
    }
}

public class HandicapAnalytics {
    public struct DifferentialEntry {
        public let score: Double
        public let courseRating: Double
    }

    public func differentials(scores: [Double]) -> [DifferentialEntry] {
        return scores.map { DifferentialEntry(score: $0, courseRating: 72.0) }
    }
}

public class FullAnalysisDataBuilder {
    public struct TimeWindow {
        public let startDay: Int
        public let endDay: Int
    }

    public func lastMonth() -> TimeWindow {
        return TimeWindow(startDay: 0, endDay: 30)
    }
}
